elastic/apm-server
# feat: add reload listener for apm tracing config

## Motivation/summary
~depending on https://github.com/elastic/beats/pull/40030~

Subscribe to the APMConfig changes from the reloader and hot-reload the runner with the updated tracing configuration.
This will enable self-instrumentation for APM Server in managed mode.

## Checklist



- [ ] Test with elastic-agent with the configuration written in `elastic-agent.yml` => will be done in separately

For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## How to test these changes
Follow instructions in `Building an Elastic Agent container image with a locally built APM Server`  and `Running an Elastic Agent container with a locally built APM Server` section in `TESTING.md`, update the `elastic-agnet.yml` to see the config is received and it's passed down to tracing correctly.


## Related issues
- #11381
- https://github.com/elastic/beats/pull/40030
